### ISSUE TITLE
Loading Knet breaks Julia's copyto! on 1.5

### ISSUE THREAD

**Issue by KristofferC** [denizyuret/Knet.jl#561]
The following methods are ambiguous with definitions in Base and thereby pretty much breaks Julia when this package is loaded:

https://github.com/denizyuret/Knet.jl/blob/886ad1e49f984ae2a2f104c858ac2420c04920e9/src/karray.jl#L322-L334

For example:

```
julia> using Knet
[ Info: Precompiling Knet [1902f260-5fb4-5aff-8c31-6271790ab950]
ERROR: LoadError: LoadError: MethodError: copyto!(::Array{String,1}, ::Int64, ::Array{String,1}, ::Int64, ::Int64) is ambiguous. Candidates:
  copyto!(dest::Array, doffs::Integer, src::Array, soffs::Integer, n::Integer) in Base at array.jl:322
  copyto!(dest::Union{Array{T,N} where N, Knet.KnetArray{T,N} where N}, doffs::Integer, src::Union{Array{T,N} where N, Knet.KnetArray{T,N} where N}, soffs::Integer, n::Integer) where T in Knet at /home/kc/.julia/packages/Knet/bTNMd/src/karray.jl:322
Possible fix, define
  copyto!(::Array{T,N} where N, ::Integer, ::Array{T,N} where N, ::Integer, ::Integer) where T
Stacktrace:
 [1] append!(::Array{String,1}, ::Array{String,1}) at ./array.jl:967
 [2] find_library(::Array{String,1}; locations::Array{String,1}, versions::Array{VersionNumber,1}, word_size::Int64) at /home/kc/.julia/packages/CUDAapi/XuSHC/src/discovery.jl:57
 [3] find_cuda_library(::String, ::Array{String,1}, ::Array{VersionNumber,1}; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/kc/.julia/packages/CUDAapi/XuSHC/src/discovery.jl:194
 [4] find_cuda_library at /home/kc/.julia/packages/CUDAapi/XuSHC/src/discovery.jl:171 [inlined]
 [5] find_toolkit() at /home/kc/.julia/packages/CUDAapi/XuSHC/src/discovery.jl:241
 [6] has_cuda() at /home/kc/.julia/packages/CUDAapi/XuSHC/src/availability.jl:24
 [7] top-level scope at /home/kc/.julia/packages/Knet/bTNMd/src/cuarray.jl:3
 [8] include(::Function, ::Module, ::String) at ./Base.jl:380
 [9] include at ./Base.jl:368 [inlined]
 [10] include(::String) at /home/kc/.julia/packages/Knet/bTNMd/src/Knet.jl:1
 [11] top-level scope at /home/kc/.julia/packages/Knet/bTNMd/src/Knet.jl:116
 [12] include(::Function, ::Module, ::String) at ./Base.jl:380
 [13] include(::Module, ::String) at ./Base.jl:368
 [14] top-level scope at none:2
 [15] eval at ./boot.jl:331 [inlined]
 [16] eval(::Expr) at ./client.jl:467
 [17] top-level scope at ./none:3
in expression starting at /home/kc/.julia/packages/Knet/bTNMd/src/cuarray.jl:3
in expression starting at /home/kc/.julia/packages/Knet/bTNMd/src/Knet.jl:116
```

**Comment by denizyuret** [comment]
Could not replicate with nightly 1.6.0-DEV.80 (is there a 1.5 I can download?)
Nevertheless pushed a fix to master bd7c162.

**Comment by KristofferC** [comment]
Yeah, https://github.com/JuliaLang/julia/pull/35965 fixed the specific ambiguity.
